configmap/parse.go

```diff
@@ -144,6 +146,33 @@ func AsQuantity(key string, target **resource.Quantity) ParseFunc {
 	}
 }
 
+// AsNamespacedName parses the value at key as a types.NamespacedName into the target, if it exists
+func AsNamespacedName(key string, target *types.NamespacedName) ParseFunc {
+	return func(data map[string]string) error {
+		raw, ok := data[key]
+		if !ok {
+			return nil
+		}
+
+		v := strings.Split(raw, string(types.Separator))
+
+		if len(v) != 2 {
+			return fmt.Errorf("failed to parse %q: expected 'namespace/name' format", key)
+		}
+
+		for _, val := range v {
+			if errs := validation.ValidateNamespaceName(val, false); len(errs) > 0 {
+				return fmt.Errorf("failed to parse %q: %s", key, strings.Join(errs, ", "))
+			}
+		}
+
+		target.Namespace = v[0]
```

Comment:
just to check: I think this requires the target to exist (otherwise this'd blow up), which means this won't support a nil default for target. I don't know if we want to support nil as a default, but it seems possible we might?

Reply:
I'd rather we have a separate method to handle the nil target case.

ie. `AsOptionalNamespacedName`

Reply:
yeah, I mean in an ideal world we create one method that works for both (like the existing AsQuantity method does), but if we don't ever see a use case for not having to supply a valid NamespacedName as the default (I don't really know the use case for passing a name to know if wanting it to be optional makes sense) then cool

Reply:
Actually in my use case it's optional so it would make sense to make it a *

Reply:
I added the optional method

Reply:
heh, cool, fwiw I think I'd personally just go with a single `AsNamespacedName` (taking a double pointer) unless we have a particular use case where we want to not support a nil default (we can always add an extra AsRequiredNamespacedName method that doesn't allow a nil default if a use case emerges for that). Right now AsOptionalNamespacedName works exactly the same as AsQuantity except it has an extra "Optional" in the name for no reason and there's an AsNamespacedName that it sounds like won't actually be used so 🤷.

(I definitely don't personally feel strongly about this, though, and this lgtm apart from that)